lib/Pipes/PipelineManager.cpp

```diff
@@ -448,3 +448,35 @@ PipelineManager::invalidateFromDiff(const llvm::StringRef Name,
   // TODO: once invalidations are working, return `Map` instead of this
   return invalidateAllPossibleTargets();
 }
+
+llvm::Error PipelineManager::produceTargets(const llvm::StringRef StepName,
+                                            const Container &TheContainer,
+                                            const TargetsList &List,
+                                            llvm::raw_ostream &Output) {
+  if (CurrentState.count(StepName) == 0)
+    return llvm::createStringError(llvm::inconvertibleErrorCode(),
+                                   "Step not found");
+
+  if (!CurrentState[StepName].contains(TheContainer.first()))
```

Comment:
same issue as before, if no target can be produced for the given container, there is no guarantee the container will apear at all in the state.

Reply:
Resolved as explained in the comment above